### ISSUE TITLE
[Feture request] post-require-cmd script

### ISSUE THREAD

**Issue by hubertnnn** [composer/composer#10890]
Currently composer has a `post-install-cmd` hook that is run after a package is installed.
This is nice for autoconfiguration tools like eg. `symfony/flex`, but is a bit problematic.

The issue is that this hook runs both after I installed manually a new package (`composer require`) 
and after a deployment script will install all packages based on `composer.lock` (`composer install`).

The effect is that `symfony/flex` is running unattended during deployment to production and either causing issues with deployment itself (flex is very often causing breaking changes on their servers, breaking all deployments) or messing with files that should be just copied from the repository and/or generated by the deployment process.

Would be nice to have separate hooks for `composer install` and `composer require`, 
so I could still benefit from autoconfiguration during installing new packages, 
but at the same time to prevent flex from running during deployments.

**Comment by nicolas-grekas** [comment]
It doesn't as long as one commits the symfony.lock file. Did you commit the file @hubertnnn? If not, that's the issue. Otherwise, I've not idea what you're referring too...

**Comment by stof** [comment]
The `post-install-cmd` script is executed only after installing from lock. `post-update-cmd` is executed after `composer update` or after an install without lock file (which is converted to an update)

**Comment by hubertnnn** [comment]
@stof 
I know, that is why I suggested `post-require-cmd`.
This way you would have granular control over scripts executed during all 3 commonly used commands:
- `composer install`  = `post-install-cmd` = install from existing composer.lock
- `composer update` = `post-update-cmd` = update all/some existing packages
- `composer require X` = `post-require-cmd` = install something new

The first 2 already exist, the third one is undocumented and if I am correct executes `post-install-cmd`

**Comment by stof** [comment]
If you don't pass the `--no-update` flag to `composer require`, it will execute `post-update-cmd` due to running a partial update.

**Comment by Seldaek** [comment]
@hubertnnn I'm still not sure what you are looking for? That we document the fact that the require command runs post-update-cmd scripts?
